helix-term/src/commands.rs

```diff
@@ -2903,11 +2901,35 @@ pub mod insert {
             ..
         }) = &capabilities.completion_provider
         {
-            // TODO: what if trigger is multiple chars long
-            if triggers.iter().any(|trigger| trigger.contains(ch)) {
-                cx.editor.clear_idle_timer();
-                super::completion(cx);
+            triggers.iter().any(|trigger| trigger.contains(ch))
+        } else {
+            false
+        }
+    }
+
+    fn language_server_completion(cx: &mut Context, ch: char) {
+        use helix_core::chars::char_is_word;
+
+        let config = cx.editor.config();
+        if !config.auto_completion {
+            return;
+        }
```

Comment:
I think we need to add one more check to show completion beside this check, it's to check whether it's running from a macro (or `repeat_last_motion`), I notice this erratic behavior of completion popping up (and sometimes not closing) when I type <kbd>.</kbd>.